internal/callbacker/dispatcher.go

```diff
@@ -87,22 +114,75 @@ func (m *sendManager) Add(token string, dto *Callback) {
 }
 
 func (m *sendManager) GracefulStop() {
-	m.wg.Wait() // wait for all accepted callbacks to be consumed
+	m.stop <- struct{}{} // signal the `run` goroutine to stop processing
+	m.wg.Wait()          // wait for all accepted callbacks to be consumed
+
 	close(m.ch) // signal the `run` goroutine to exit
+
+	<-m.stop // wait for the `run` goroutine to exit
+	close(m.stop)
 }
 
 func (m *sendManager) run() {
 	go func() {
```

Comment:
Don't we also need to add 1 to the wait group here? 🤔

Reply:
this wait group is used for accepted entries (callbacks), so it is modified in the `Add()` method.